code/game/objects/structures/displaycase.dm

```diff
@@ -20,14 +20,18 @@
 	///If we have a custom glass overlay to use.
 	var/custom_glass_overlay = FALSE
 	var/obj/item/electronics/airlock/electronics
-	///add type for items on display
+	///Add type for items on display
 	var/start_showpiece_type = null
-	///displaycase is fixed by glass
+	///Displaycase is fixed by glass
 	var/glass_fix = TRUE
 	///Represents a signel source of screaming when broken
 	var/datum/alarm_handler/alarm_manager
 	///Used for subtypes that have a UI in them. The examine on click while adjecent will not fire, as we already get a popup
 	var/autoexamine_while_closed = TRUE
+	///Does the display case uses a UI that requires caching?
+	var/uses_showpiece_cache = FALSE
```

Comment:
What doesn't want to cache?

Reply:
Regular display cases that have no UI.

Reply:
Instead of doing it like this, instead create the cache on ui_data/ui_static_data.

Reply:
Ah. That is indeed much better that way.